### PR TITLE
Added progress bar for the quiz 

### PR DESCRIPTION
**Pull request by Jaykovsky**
Hi, 

I made some some changes and i fixed the ProgressBar component. 
Now it displays the progress correctly.

The new component is called `ProgressBar.jsx`, it received two opcional props: `progressBarColor` and `height`.
You can enable or disable with the prop: `enableProgressBar` (Its receive a boolean value) at the Quiz component; and you can change the color of the progress bar with the prop: `progressBarColor` (It receive a string value).

**First question**:

**Second question**:

**Third question**:

**Four question**:

**End quiz**:

### CODE REVIEW

File: src/lib/Quiz.jsx [hunk 1]
```diff
@@ -108,6 +110,16 @@ function Quiz({
       return false;
     }
 
+    if (enableProgressBar && typeof enableProgressBar !== 'boolean') {
+      console.error('enableProgressBar must be a Boolean');
+      return false;
+    }
+
+    if (progressBarColor && typeof progressBarColor !== 'string') {
```
Comment (wingkwong) [inline src/lib/Quiz.jsx:118]: what if i input a not-color code? e.g. `abc`

**Review by wingkwong — COMMENTED** [review]
please also include your new changes in `props` in  README.md

https://github.com/wingkwong/react-quiz-component?tab=readme-ov-file#props

File: src/lib/Quiz.jsx (hunk 1 above)
Comment (Jaykovsky) [reply src/lib/Quiz.jsx:123]: Hi, i added a function that verifies the hex value for the progress bar.

**Comment by Jaykovsky** [comment]
> please also include your new changes in `props` in README.md
> 
> https://github.com/wingkwong/react-quiz-component?tab=readme-ov-file#props

It's done.

**Review by wingkwong — CHANGES_REQUESTED** [review]
Somehow I think it's a bit redundant to have two props in order to show the progress bar.  We can do something as follows:

- enable the progress bar only `enableProgressBar` is `true` (which you've already done that)
- set a default `progressBarColor`. However, users can override the color by specifying in  `quiz` object. e.g.

```jsx
const quiz = {
  quizTitle: 'React Quiz Component Demo',
  quizSynopsis: 'Lorem ipsum dolor',
  nrOfQuestions: '6',
  questions: [{...}],
  progressBarColor: '#FF0000'
};
```

```js
<Quiz
  quiz={quiz}
  enableProgressBar
/>
```

**Comment by Jaykovsky** [comment]
I got you bro.

The changes are already made. U can check and review.

File: src/lib/Quiz.jsx [hunk 2]
```diff
@@ -108,6 +114,21 @@ function Quiz({
...
+
+    if (quiz.progressBarColor && typeof quiz.progressBarColor !== 'string') {
+      console.error('progressBarColor must be a String');
+      return false;
+    }
+
+    if ('progressBarColor' in quiz && !validateProgressBarColor(quiz.progressBarColor)) {
+      console.error('progressBarColor must be a valid hex colour');
+      return false;
+    }
```
Comment (wingkwong) [inline src/lib/Quiz.jsx:130]:
```jsx
if ('progressBarColor' in quiz) {
  if (typeof quiz.progressBarColor !== 'string') {
      console.error('progressBarColor must be a String');
      return false;
  }
  if (!validateProgressBarColor(quiz.progressBarColor)) {
      console.error('progressBarColor must be a valid hex colour');
      return false;
  }
}
```

File: README.md [hunk 3]
```diff
@@ -47,6 +47,7 @@ The quiz source is a JSON object. You can use [react-quiz-form](https://github.c
 export const quiz =  {
   "quizTitle": "React Quiz Component Demo",
   "quizSynopsis": "Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim",
+  //"progressBarColor": "#FF0000",
```
Comment (wingkwong) [inline README.md:50]: uncomment it

File: README.md [hunk 4]
```diff
@@ -295,6 +305,7 @@ import { quiz } from './quiz';
 |disableSynopsis|`boolean`|`false`|N|Disable synopsis before quiz|
 |timer|`number`|`false`|N|Sets timer in seconds|
 |allowPauseTimer|`boolean`|`false`|N|Pause / Resume timer|
+|enableProgressBar|`boolean`|`true`|N|Enables a progress bar|
```
Comment (wingkwong) [inline README.md:308]: 1. set the default value to `false`
2. update the description to the following

```
Enable a progress bar
```

File: src/docs/quiz.jsx [hunk 5]
```diff
@@ -8,6 +8,7 @@ const quiz = {
   quizTitle: 'React Quiz Component Demo',
   quizSynopsis: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim',
   nrOfQuestions: '6',
+  //progressBarColor: '#FF0000',
```
Comment (wingkwong) [inline src/docs/quiz.jsx:11]: uncomment it

File: src/lib/core-components/ProgressBar.jsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+      <span style={progressBarLabel}>
+        {isEndQuiz ? '100%' : `${Math.round(progressUnit * fixedProgress)}%`}
+      </span>
+    </div>
+  );
+};
+
+ProgressBar.defaultProps = {
+  progressBarColor: '#9de1f6', // Set a default value for bgcolor
+  height: '25px', // Set a default value for bar's heigth
```
Comment (wingkwong) [inline src/lib/core-components/ProgressBar.jsx:47]: typo in comment: `height`

File: src/lib/core-components/ProgressBar.jsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+      <div style={progressBar} />
+      <span style={progressBarLabel}>
+        {isEndQuiz ? '100%' : `${Math.round(progressUnit * fixedProgress)}%`}
+      </span>
+    </div>
+  );
+};
+
+ProgressBar.defaultProps = {
+  progressBarColor: '#9de1f6', // Set a default value for bgcolor
```
Comment (wingkwong) [inline src/lib/core-components/ProgressBar.jsx:46]:
```
Set a default value for background color
```

**Review by wingkwong — CHANGES_REQUESTED** [review]
should be final round of changes

**Comment by Jaykovsky** [comment]
@wingkwong All requested changes have been applied.

File: src/lib/Quiz.jsx [hunk 8]
```diff
@@ -108,6 +114,23 @@ function Quiz({
       return false;
     }
 
+    if (enableProgressBar && typeof enableProgressBar !== 'boolean') {
+      console.error('enableProgressBar must be a Boolean');
+      return false;
+    }
+
+    if ('progressBarColor' in quiz) {
+      if (typeof quiz.progressBarColor !==) {
```
Comment (wingkwong) [inline src/lib/Quiz.jsx:123]: broken

**Review by wingkwong — CHANGES_REQUESTED** [review]
please test it once before requesting review.

**Comment by Jaykovsky** [comment]
@wingkwong Sorry, i forget to test before push. 

Now the error is gone, i tested. U can check.

**Pull request merged**